src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -591,6 +590,12 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			this._xtermReadyPromise.then(xterm => {
 				contribution.xtermReady?.(xterm);
 			});
+			this._containerReadyBarrier.wait().then(() => {
+				if (!this.xterm) {
+					throw new Error('Container is ready but xterm is not');
+				}
+				contribution.layout?.(this.xterm);
+			});
```

Comment:
We would loop over all contributions in `layout` and do it there, since we want to call for every layout, not just the first.